bridge-impl/src/main/java/com/liferay/faces/bridge/scope/internal/BridgeRequestScopeManagerImpl.java

```diff
@@ -39,105 +43,164 @@
 	// Logger
 	private static final Logger logger = LoggerFactory.getLogger(BridgeRequestScopeManagerImpl.class);
 
-	public void removeBridgeRequestScopesByPortlet(PortletConfig portletConfig) {
-		String portletNameToRemove = portletConfig.getPortletName();
-		PortletContext portletContext = portletConfig.getPortletContext();
-		BridgeRequestScopeCache bridgeRequestScopeCache = BridgeRequestScopeCacheFactory
-			.getBridgeRequestScopeCacheInstance(portletContext);
-		Set<Map.Entry<String, BridgeRequestScope>> mapEntries = bridgeRequestScopeCache.entrySet();
+	// Private Constants
+	private static final String ATTR_BRIDGE_REQUEST_SCOPE_CACHE = "com.liferay.faces.bridge.bridgeRequestScopeCache";
+
+	@Override
+	public Cache<String, BridgeRequestScope> getBridgeRequestScopeCache(PortletContext portletContext) {
+
+		Cache<String, BridgeRequestScope> bridgeRequestScopeCache;
+
+		synchronized (portletContext) {
```

Comment:
@stiemannkj1 Should we be using the double-check idiom here? I think we probably should since this gets called from BridgePhaseBaseImpl during the scope of a request.